std/math.d

```diff
@@ -7291,3 +7292,85 @@ unittest
         }
     }
 }
+
+/**
+Gives the next highest power of two after $(D val). If the input would result
+in an overflow, then zero is returned.
+
+Params:
+    val = a $(D size_t)
+
+Returns:
+    the next highest power of two after $(D val) as a $(D size_t)
+*/
+size_t nextPow2(const size_t val) pure nothrow @nogc @safe
+{
+    if (val == 0)
+    {
+        return 1;
+    }
+
+    enum max = size_t.max / 2;
+    if (val <= max)
+    {
+        size_t res = 1;
+        while (res < val)
+        {
+            res <<= 1;
+        }
+
+        return res;
+    }
+    else
+    {
+        return 0;
+    }
+
+}
+
+///
+unittest
+{
+    assert(nextPow2(1) == 1);
+    assert(nextPow2(2) == 2);
+    assert(nextPow2(10) == 16);
+    assert(nextPow2(4000) == 4096);
+
+    assert(nextPow2(size_t.max) == 0);
+}
+
+unittest
+{
+    version(X86)
+        immutable arr = [4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048,
+        4096, 8192, 16384, 32768, 65536, 131072, 262144, 524288, 1048576,
+        2097152, 4194304, 8388608, 16777216, 33554432, 67108864, 134217728,
+        268435456, 536870912, 1073741824, 2147483648];
+    else
+        immutable arr = [4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048,
+        4096, 8192, 16384, 32768, 65536, 131072, 262144, 524288, 1048576,
+        2097152, 4194304, 8388608, 16777216, 33554432, 67108864, 134217728,
+        268435456, 536870912, 1073741824, 2147483648, 4294967296, 8589934592,
+        17179869184, 34359738368, 68719476736, 137438953472, 274877906944,
+        549755813888, 1099511627776, 2199023255552, 4398046511104,
+        8796093022208, 17592186044416, 35184372088832, 70368744177664,
+        140737488355328, 281474976710656, 562949953421312, 1125899906842624,
+        2251799813685248, 4503599627370496, 9007199254740992,
+        18014398509481984, 36028797018963968, 72057594037927936,
+        144115188075855872, 288230376151711744, 576460752303423488,
+        1152921504606846976, 2305843009213693952, 4611686018427387904];
```

Comment:
You could've at least used binary or hexadecimal base constants to make this list readable.